core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -31,12 +37,13 @@
  */
 public class TermSuggestionBuilder extends SuggestionBuilder<TermSuggestionBuilder> {
 
+    public static final TermSuggestionBuilder PROTOTYPE = new TermSuggestionBuilder("_na_"); // name doesn't matter
     static final String SUGGESTION_NAME = "term";
 
-    private String suggestMode;
+    private SuggestMode suggestMode;
     private Float accuracy;
-    private String sort;
-    private String stringDistance;
+    private SortBy sort;
+    private StringDistanceImpl stringDistance;
     private Integer maxEdits;
     private Integer maxInspections;
     private Float maxTermFreq;
```

Comment:
These should all have sensible defaults rather than being null, as we have done with all the other builders. This also means we can remove @Nullable from all of the methods and add checks in there that throw an exception if null is passed in to make it safer. The defaults we currently use can be found in DirectSpellcheckerSettings.